lib/hsa/mcwamp_hsa.cpp

```diff
@@ -1529,10 +1529,15 @@ class HSAQueue final : public KalmarQueue
                 if (FORCE_SIGNAL_DEP_BETWEEN_COPIES) {
                     DBOUT(DB_CMD2, "Set NeedDep for " << newOp << "(FORCE_SIGNAL_DEP_BETWEEN_COPIES) " );
                     needDep = true;
+                    return lastOp;
```

Comment:
Why would we return lastOp here?  Why would we force the signal dep between copies even if the signal is already complete (as determined by line 1537 below)?

Reply:
Fixed